src/util/sock.h

```diff
@@ -155,6 +155,8 @@ class Sock
      * Contained socket. `INVALID_SOCKET` designates the object is empty.
      */
     SOCKET m_socket;
+
+    friend class FuzzedSock;
```

Comment:
style-nit in the first commit: Wouldn't it be better to change `private` to `protected`? This would allow other tests to mock `m_socket` without having to enumerate all "friend" test classes here

Reply:
Done.